frontend/src/api/actions/init.ts

```diff
@@ -181,12 +181,8 @@ export function initFromStage(
         if (shouldRunStage("setActiveUserRole")) {
             // When the role is changed, data should be cleared immediately.
             // It will be re-fetched via the appropriate routes.
-            dispatch(fetchApplicantsSuccess([]));
-            dispatch(fetchAssignmentsSuccess([]));
-            dispatch(fetchContractTemplatesSuccess([]));
-            dispatch(fetchApplicationsSuccess([]));
+            dispatch(clearSessionDependentData());
             dispatch(fetchInstructorsSuccess([]));
```

Comment:
Do we need to await these? Does `setActiveUserRole` change state?

Reply:
I don't think instructors depend on a session. There is no need to clear them.

Reply:
Sorry, not sure I understand. That's not session dependent data indeed, but I suppose we are clearing instructors in `setActiveUserRole` because we might be changing the permissions and shouldn't have them in a state anymore?

Reply:
Ah. I see. This is not a line you added :-)